src/wamr/CMakeLists.txt

```diff
@@ -5,40 +5,30 @@
 # We have our own custom CMake set-up which looks a bit like the wamrc build:
 # https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/wamr-compiler/CMakeLists.txt
 
-# Turn on WAMR features
```

Comment:
@Shillaker this has changed since your first review.

In short, adding the stricter spec checks globally made the compilation here fail. I took the chance to then clean the wamr build the same way it is done for SGX.

In the process, I realised that we are relying on `LIBC_WASI` for `WAMR` (non-SGX) builds. Removing this dependency means that we need to add quite some function definitions (like `fd_close`, `fd_seek`, `fd_write`, ...). I defer this to a future PR, but soon enough as it will easily fall in oblivion.

Reply:
👍